DOC-3774: Docker docs improvements (#561)
corda/corda-docs-portal: content/en/platform/corda/4.8/enterprise/node-docker-deployments.md

        -p 10200:10200 \
        -p 10201:10201 \
        corda/corda-zulu-java1.8-4.8:latest
```

The mount `/home/user/sharedFolder/node-infos:/opt/corda/additional-node-infos` is used to hold the `nodeInfo` of all the nodes within the network.
As the node within the container starts up, it will place its own nodeInfo into this directory. This will allow other nodes also using this folder to see this new node.

## Generating configs and certificates

It is possible to utilize the image to automatically generate a sensible minimal configuration for joining an existing Corda network.

## Joining an existing Compatibility Zone

{{< note >}}
Requirements: A Compatibility Zone, the Zone Trust Root and authorisation to join said Zone.

{{< /note >}}
It is possible to use the image to automate the process of joining an existing Zone as detailed [here](../../../../../en/platform/corda/4.8/enterprise/network/compatibility-zones.md)

The first step is to obtain the Zone Trust Root, and place it within a directory. In the below example, the Trust Root is stored at `/home/user/docker/certificates/network-root-truststore.jks`.

> It is possible to configure the name of the Trust Root file by setting the `TRUST_STORE_NAME` environment variable in the container.

Reviewer: ```suggestion
You can configure the name of the Trust Root file by setting the `TRUST_STORE_NAME` environment variable in the container.
```

        -e SSHPORT="2222"      \
        -e RPC_USER="PartyA"      \
        -v /home/user/docker/config:/etc/corda          \
        -v /home/user/docker/certificates:/opt/corda/certificates \
        corda/corda-zulu-java1.8-4.8:latest config-generator --generic --exit-on-generate
```

Several environment variables must also be passed to the container to allow it to register:

* `MY_LEGAL_NAME` - The X500 to use when generating the config. This must be the same as registered with the Zone.
* `MY_PUBLIC_ADDRESS` - The public address to advertise the node on.
* `NETWORKMAP_URL` - The address of the Zone’s network map service (this should be provided to you by the Zone).
* `DOORMAN_URL` - The address of the Zone’s doorman service (this should be provided to you by the Zone).
* `NETWORK_TRUST_PASSWORD` - The password to the Zone Trust Root (this should be provided to you by the Zone).
* `MY_EMAIL_ADDRESS` - The email address to use when generating the config. This must be the same as registered with the Zone.
* `SSHPORT` - The port to use for SSH.
* `RPC_USER` - The RPC username.

There are some optional variables which allow customisation of the generated config:

> 

Reviewer: please can you squish these double line breaks.

* `MY_P2P_PORT` - The port to advertise the node on (defaults to 10200). If changed, ensure the container is launched with the correct published ports.
* `MY_RPC_PORT` - The port to open for RPC connections to the node (defaults to 10201). If changed, ensure the container is launched with the correct published ports.

> Once the container has finished performing the initial registration, the node can be started as normal

Reviewer: This sentence has no ending.

        -v /home/user/docker/persistence:/opt/corda/persistence \
        -v /home/user/docker/logs:/opt/corda/logs \
        -v /home/user/corda/samples/bank-of-corda-demo/build/nodes/BankOfCorda/cordapps:/opt/corda/cordapps \
        -p 10200:10200 \
        -p 10201:10201 \
        corda/corda-zulu-java1.8-4.8:latest
```

# Performing Database Migrations

The image contains the database-migration tool. It is possible to run this in two modes within a container.

## Generating Migration Jars

In this mode, the database-migration tool will scan the provided CorDapps, and generate corresponding migration jars. These jars will be placed alongside
the source CorDapps. In this example, there are two CorDapps provided `corda-insurance.jar` and `corda-kyc.jar`

```shell
docker run -ti \
        -v /home/user/docker/docker/config:/etc/corda \
        -v /home/user/docker/docker/certificates:/opt/corda/certificates \
        -v /home/user/docker/docker/persistence:/opt/corda/persistence \
        -v /home/user/docker/docker/logs:/opt/corda/logs \
        -v /home/user/corda/samples/bank-of-corda-demo/build/nodes/BankOfCorda/cordapps:/opt/corda/cordapps \

>         entdocker.software.r3.com/corda-enterprise-java1.8-4.8:latest db-migrate-create-jars

Reviewer 2: The  image entdocker.software.r3.com/corda-enterprise-java1.8-4.8:latest no longer seems to exist.
However, corda/corda-enterprise-alpine-zulu-java1.8-4.8.6:latest, from docker hub does.

```

After the container has finished executing, there will be two new jars in `/home/user/corda/samples/bank-of-corda-demo/build/nodes/BankOfCorda/cordapps`: `migration-corda-insurance.jar` and `migration-corda-kyc.jar`.
These will then be loaded as normal CorDapps by the node on next launch.

## Executing Migrations on the Database

It is also possible to use the image to directly perform the migration of the database.

```shell
docker run -ti \
        -v $(pwd)/docker/config:/etc/corda \
        -v $(pwd)/docker/certificates:/opt/corda/certificates \
        -v $(pwd)/docker/persistence:/opt/corda/persistence \
        -v $(pwd)/docker/logs:/opt/corda/logs \
        -v $(pwd)/samples/bank-of-corda-demo/build/nodes/BankOfCorda/cordapps:/opt/corda/cordapps \

>         entdocker.software.r3.com/corda-enterprise-java1.8-4.8:latest db-migrate-execute-migration

Reviewer 2: Image no longer seems to exist seem comment above.
